File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/subsystems/Intake.java
package org.firstinspires.ftc.teamcode.drive.subsystems;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.CRServo;

public class Intake {
    private CRServo Intake;

    public Intake(HardwareMap hardwareMap) {
        this.Intake = hardwareMap.get(CRServo.class, "Intake");
        this.Intake.setDirection(CRServo.Direction.FORWARD);
    }
    public void setPower(double power) {
        Intake.setPower(power);
    }
}

File: TeamCode/src/main/kotlin/org/firstinspires/ftc/teamcode/botmodule/Intake.kt
package org.firstinspires.ftc.teamcode.botmodule

import com.qualcomm.robotcore.eventloop.opmode.OpMode
import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.DcMotorEx
import com.qualcomm.robotcore.hardware.Servo

/**
 * Intake controls
 */
@Suppress("unused")
class Intake(opMode: OpMode, private val motorLift: Servo?, private val motorSpin: DcMotorEx?) : BotModule(opMode) {
    init {
        /*
        motorLift?.targetPosition = 0
        motorLift?.mode = DcMotor.RunMode.RUN_TO_POSITION
        motorSpin?.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        motorSpin?.power = 0.0
        motorLift?.power = 0.2
//        servoRight

         **/
    }

    // 0.0 is off, 1.0 is inwards, -1.0 is outwards
    var active: Double = 0.0
        set(status) {
            if (motorSpin == null) field = 0.0
            else {
                field = status
                motorSpin.power = if (field > 1.0) 1.0 else if (field < -1.0) -1.0 else field
            }
        }
    var raised: Boolean = false
        private set

    fun lower() {
        motorLift?.position = .5
        raised = false
    }

    fun raise() {
        motorLift?.position = .1
        raised = true
    }
}